src/content/blog/ddev-name-resolution-wildcards.md

```diff
@@ -0,0 +1,99 @@
+
+DDEV supports wildcards in `additional_hostnames` in your `.ddev/config.yaml`. For example, this works great when you have internet and DNS working, because DDEV's `.ddev.site` will return `127.0.0.1` for any domain name you give it, including `a.b.c.d.e.f.g.h.ddev.site`.  So if you use `additional_hostnames: *.anything` then `a.anything.ddev.site` will work fine, as will `a.b.anything.ddev.site`. 
+
+However, wildcards don't work in `/etc/hosts` so DDEV has no way to provide a resolvable set of hostnames when you don't have internet work working DNS. As a result, you'll need to use explicit hostnames in this situation. For example, use `additional_hostnames: [a.anything,a.b.anything]`
+
+## Can I stop using the regular DNS setup with `ddev.site`?
+
+Of course. There are two easy ways:
+1. `use_dns_when_possible: false`: Turn off the use of DNS in your `.ddev/config.yaml`.
+2. Use a different `project_tld`. For example, if you use `project_tld: example.site` then DDEV will try to make your projects `<projectname>.example.site`. Since that is not resolvable, DDEV will resort to using hosts file manipulation.
+
+
+## Can I use my own local-machine DNS?
+
+Yes, you can set up a DNS provider on your own computer. [blocky](https://github.com/0xERR0R/blocky) and [dnsmasq](https://thekelleys.org.uk/dnsmasq/doc.html) are traditional choices. I don't recommend this for most people, because it means you need to understand much more about networking and extra software than you would normally need to know. But if you do this, you can provide wildcard services that you're in control of and that work even when your connection to the internet is broken.
+
+## Can I set up a domain name on my local network or company network?
+
+Yes, this is exactly the same. Normally you would have every DDEV-related domain name return `127.0.0.1` for the DNS name you're supporting. However, some people use this technique to [share DDEV sites inside a local network](https://ddev.readthedocs.io/en/stable/users/topics/sharing/#exposing-a-host-port-and-providing-a-direct-url). 
+
+## Can I set up my own domain in DNS like `ddev.site`?
+
+Yes, if you have the ability to configure an internet DNS zone, you can set it up the same way `ddev.site` is set up, resolving to `127.0.0.1` and with a wildcard that resovles to `127.0.0.1`.
+
+## What happens on WSL2 and why do I have to take action on the Windows side?
+
+On WSL2, when you don't have internet, or your DNS is broken, or you're using a hostname that can't be looked up in DNS, DDEV has to punt the hosts file change over to the Windows side of things, because your *browser* is almost always running on Windows, so editing the hosts file on the Windows side is what has to happen. The way DDEV currently does this is requests a **Windows** version of `ddev.exe` to escalate and edit the hosts file. That's why WSL2 installations also ask you to install Windows DDEV using `choco install -y ddev`.
```

Comment:
"DDEV has to punt the hosts file change over to the Windows side of things"
i dont have a suggestion but that part i can barley grasp what it means because i dont know what file change means in this context (plus i not exactly know what punt means. plus the sentence is lengthy. it could be easily split up into at least two. 

"The way DDEV currently does this is requests a **Windows** version of `ddev.exe` to escalate and edit the hosts file." 
that i also dont fully understand. does it mean. what does requesting a ddev.exe mean? is that compile? and escalating means the equivalent to sudo in the unix world?

Reply:
```suggestion
On WSL2, when you don't have internet, or your DNS is broken, or you're using a hostname that can't be looked up in DNS, DDEV has to request the hosts file change from the Windows side of things, because your *browser* is almost always running on Windows, so editing the hosts file on the Windows side is what has to happen. The way DDEV currently does this is requests a **Windows** version of `ddev.exe` to escalate and edit the hosts file. That's why WSL2 installations also ask you to install Windows DDEV using `choco install -y ddev`.
```